hermes/bin/src/runtime_extensions/hermes/crypto/bip39.rs

```diff
@@ -0,0 +1,392 @@
+    // Convert bits_entropy to bytes, so it works with SHA256 hasher.
+    let bytes_entropy = bits_to_bytes(entropy_bits);
+    hasher.input(&bytes_entropy);
+
+    hasher.result(&mut hash_result);
+
+    // Retrieve the first `check_sum_num` check sum bits from the hash result.
+    let mut check_sum_bits = Vec::new();
+    for i in 0..check_sum_num {
+        check_sum_bits.push(hash_result[0] >> (7 - i) & 1);
+    }
+    check_sum_bits
+}
+
+/// Get the word indices from the entropy bits.
+fn get_word_indices(bits_entropy: &[u8], word_count: usize) -> Vec<u16> {
+    let mut word_index_vec = Vec::new();
+
+    // Separate entropy bits into 11 bits and convert to decimal.
+    // This decimal will be used to get the word index.
+    for i in 0..word_count {
+        let mut idx = 0;
+        for j in 0..11 {
+            if let Some(value) = bits_entropy.get(i * 11 + j) {
+                if *value > 0 {
+                    idx += 1 << (10 - j);
+                }
+            }
+        }
+        word_index_vec.push(idx);
+    }
+    word_index_vec
+}
+
+/// Get the mnemonic from the BIP39 dictionary using word indices.
+fn get_mnemonic_from_indices(word_index_vec: Vec<u16>, language: Language) -> Vec<String> {
+    let word_list = language.word_list();
+    let mut mnemonic: Vec<String> = vec![];
+    for word in word_index_vec {
+        if let Some(word) = word_list.get(word as usize) {
+            mnemonic.push((*(word)).to_string());
+        }
+    }
+    mnemonic
+}
+
+/// Turns decimal into binary array of length 11.
+fn decimal_to_binary_array(decimal: u16) -> [u8; 11] {
+    let mut binary = [0u8; 11];
+    let mut n = decimal;
+    let mut index = 0;
+
+    while n > 0 {
+        if let Some(value) = binary.get_mut(index) {
+            let bit = n % 2;
+            index += 1;
+            *value = bit as u8;
+            n /= 2;
+        }
+    }
+
+    // If the number of bits is less than 11, fill the remaining bits with 0.
+    while index < 11 {
+        if let Some(value) = binary.get_mut(index) {
+            *value = 0;
+        }
+        index += 1;
+    }
+    binary.reverse();
+    binary
+}
+
+/// Turns bit of array to bytes of arrays
+fn bits_to_bytes(bits: &[u8]) -> Vec<u8> {
+    bits.chunks(8)
+        .map(|chunk| {
+            chunk
+                .iter()
+                .enumerate()
+                .fold(0, |acc, (i, &bit)| acc | ((bit) << (7 - i)))
+        })
+        .collect()
+}
+
+/// Convert string to BIP39 language.
+fn string_to_language(s: &str) -> Language {
+    match s {
+        "SimplifiedChinese" => Language::SimplifiedChinese,
+        "TraditionalChinese" => Language::TraditionalChinese,
+        "Czech" => Language::Czech,
+        "French" => Language::French,
+        "Italian" => Language::Italian,
+        "Japanese" => Language::Japanese,
+        "Korean" => Language::Korean,
+        "Spanish" => Language::Spanish,
+        _ => Language::English,
```

Comment:
Don't put English as default here. Think like you're implementing the `FromStr` trait, you always need to return with `Result<T, E>`.